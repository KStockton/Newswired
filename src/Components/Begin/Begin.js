import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux';



export class Begin extends Component {
  static propTypes = {
    prop: PropTypes
  }

  render() {
    return (
      <div className="begin-wrapper">
        <div >
        <h1>Welcome to News Wire</h1>  
        </div>
        <button className="begin-btn">Get Started</button>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  
})

const mapDispatchToProps = {
  
}

export default connect(mapStateToProps, mapDispatchToProps)(Begin)
