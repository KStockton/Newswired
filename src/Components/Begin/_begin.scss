@import '../../Sass/_variables';

.begin-wrapper {
  background-image: url('../Assets/splash.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  margin: 0;
  position: fixed;
  width: 100vw;
  background-position-y: center;
}

#welcome{
  font-size: 16vw;
  font-family: 'Old Standard TT', serif;
  margin: auto;
  position: fixed;
  right: 25%;
  top: 25%;
  color: white;
}

#news-wire{
  font-size: 3rem;
  margin: auto;
  position: fixed;
  top: 57%;
  right: 12%;
  color: #837a7a;
}

.start{
  position: fixed;
  bottom: 22%;
  right: 14%;
  margin: auto;
  z-index: 2;
  cursor: pointer;
}

.begin-btn {
  padding: 10px;
  position: relative;
  text-decoration: none;
  color: #000;
  border: solid 1px #186f8f;
  border-radius: 5px;
  text-transform: uppercase;
  z-index: auto;
  box-shadow: 3px 3px 2px #52000000;
}
.begin-btn:hover{
    padding-left: 10px;
    padding-right: 10px;
    padding-top: 11px;
    background-image: -webkit-gradient(linear, left bottom, left top, color-stop(0, #2cffc9), color-stop(1, #ffffff));
    color: white;
    background: black;
    box-shadow: inset -1px 0px 8px 2px white;
    cursor: pointer;
}

.begin-btn::before {
  background-color: #ccd0d56b;
  content: "";
  display: block;
  position: absolute;
  width: 100%;
  height: 100%;
  padding: 8px;
  left: -8px;
  top: -8px;
  border-radius: 5px;
  box-shadow: inset 0px 1px 1px #909193, 0px 1px 0px #fff;
}

.begin-btn:active {
  padding-bottom: 9px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 11px;
  top: 1px;
  background-image: -webkit-gradient( linear, left bottom, left top, color-stop(0, rgb(62,184,229)), color-stop(1, rgb(44,160,202)) );
}

@media screen and (max-width: 1200px){

  #welcome{
    position: fixed;
    right: 25%;
    top: 25%;
  }
}

@media screen and (max-width: 1100px) {

  .begin-wrapper {
    background-position-x: 50%;
  }
}

@media screen 
and (max-width: 980px)
and (max-width: 760px) {

  #welcome-container {
    display: flex;
    align-content: center;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    height: 100%;
  }
  
  
  #welcome, #news-wire, .start {
    margin: 0;
    position: static;
  }
  #news-wire {
    font-size: 6vw;
    background-color: white;
  }

  .begin-btn {
    margin-top: 179px;
  }
}

@media screen and (max-width: 760px){

  .begin-wrapper {
    // background-position-x: 25%;
  }

  #welcome{
    // font-size: 4em;
    // position: fixed;
    // right: 23%;
    // top: 30%;
  }

}

// @media screen and (max-width: 760px){

//   .begin-wrapper {
//     background-position-x: 40%;
//   }

//   #welcome{
//     font-size: 3.5em;
//     position: fixed;
//     right: 23%;
//     top: 25%;
//   }

//   #news-wire{
//     font-size: 3rem;
//     position: fixed;
//     top: 45%;
//   }

//   .start{
//     position: fixed;
//     bottom: 29%;
//   }
// }

// @media screen and(max-width: 620px){

//   .begin-wrapper {
//     background-position-x: 48%;
//   }

//   #welcome{
//     font-size: 3em;
//     position: fixed;
//     right: 18%;
//     top: 25%;
//   }

//   #news-wire{
//     font-size: 2.8rem;
//     position: fixed;
//     top: 37%;
//     right: 19%;
//   }

//   .start{
//     position: fixed;
//     bottom: 40%;
//     right: 35%;
//   }
  
// }

@media screen and(max-width: 320px){

  // #welcome{
  //   font-size: 1.5em;
  // }

  // #news-wire{
  //   font-size: 1.5rem;
  //   top: 35%;
  // }

  // .start{
  //   font-size: 14px;
  //   bottom: 16%;
  //   right: 32%;
  // }
}