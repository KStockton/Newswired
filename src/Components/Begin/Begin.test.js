import React from 'react';
import Begin from './Begin';
import { shallow } from 'enzyme';

describe('Begin', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Begin />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})