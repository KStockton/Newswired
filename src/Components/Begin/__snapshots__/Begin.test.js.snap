// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Begin should match the snapshot 1`] = `
<div
  className="begin-wrapper"
>
  <section>
    <h1>
      <span
        id="welcome"
      >
        Welcome
      </span>
      <span
        id="news-wire"
      >
        to News Wire
      </span>
    </h1>
  </section>
  <Link
    className="start"
    to="/Options"
  >
    <button
      className="begin-btn"
    >
      Get Started
    </button>
  </Link>
</div>
`;
